### DISCUSSION TITLE
What do you think unmounted Atom GC automatically ？

### DISCUSSION THREAD

**Discussion by SquirrelJimmy** [pmndrs/jotai#499] [Ideas]
As the title, if some atom had been unmounted, delete it from global state,
`state.a.delete(unmountedAtom)`;

In source code 
```ts
const unmountAtom = <Value>(state: State, atom: Atom<Value>): void => {
// ...
  const atomState = getAtomState(state, atom)
  if (atomState) {
    atomState.d.forEach((_, a) => {
      if (a !== atom) {
        const mounted = state.m.get(a)
        if (mounted) {
          mounted.d.delete(atom)
          if (canUnmountAtom(a, mounted)) {
            unmountAtom(state, a)
          }
        }
      }
    })
    state.a.delete(atom)
  }
// ...
}
```
if do nothing, the state is always growing though some Atom is not used

**Comment by dai-shi** [comment]
We can't delete atom values immediately after unmounted, because there are use cases that depend on persisted atoms values after remount.
Jotai stores atom values in WeakMap, so unless you keep the atom config objects in memory, they will be garbage collected eventually. And, keeping the atom config basically means you want to remount the atom later (or memory leak).
Ref: https://github.com/pmndrs/jotai/issues/420#issuecomment-819462695
